### PR TITLE
[SPARK-37237][K8S][BUILD] Upgrade kubernetes-client to 5.9.0

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to upgrade `kubenetes-client` from 5.8.0 to 5.9.0.

### Why are the changes needed?

This will bring the following bug fixes and improvement.
- https://github.com/fabric8io/kubernetes-client/releases/tag/v5.9.0

### Does this PR introduce _any_ user-facing change?

No. (The supported K8s versions are identical with 5.8.0).

To make Scala 2.13 compilation pass, the new `Action` type, `Bookmark`, is handled in the test case.
- https://github.com/fabric8io/kubernetes-client/pull/3488

### How was this patch tested?

Pass the CIs.

### CODE REVIEW

**Review by sarutak — APPROVED** [review]
LGTM. Pending CIs.

**Comment by dongjoon-hyun** [comment]
Thank you, @sarutak !

**Comment by dongjoon-hyun** [comment]
Although the UT passed, K8s integration seems to hit Jackson incompatibility issue. I'll take a look

```
KubernetesSuite:
*** RUN ABORTED ***
  java.lang.NoSuchMethodError: com.fasterxml.jackson.dataformat.yaml.YAMLFactory._createContentReference(Ljava/lang/Object;)Lcom/fasterxml/jackson/core/io/ContentReference;
  at com.fasterxml.jackson.dataformat.yaml.YAMLFactory.createParser(YAMLFactory.java:374)
  at com.fasterxml.jackson.dataformat.yaml.YAMLFactory.createParser(YAMLFactory.java:348)
  at com.fasterxml.jackson.dataformat.yaml.YAMLFactory.createParser(YAMLFactory.java:15)
  at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3548)
  at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3516)
  at io.fabric8.kubernetes.client.internal.KubeConfigUtils.parseConfigFromString(KubeConfigUtils.java:47)
  at io.fabric8.kubernetes.client.Config.loadFromKubeconfig(Config.java:591)
  at io.fabric8.kubernetes.client.Config.tryKubeConfig(Config.java:557)
  at io.fabric8.kubernetes.client.Config.autoConfigure(Config.java:279)
  at io.fabric8.kubernetes.client.Config.<init>(Config.java:245)
```

**Comment by dongjoon-hyun** [comment]
This seems to be the root cause at 2.13.0
- https://github.com/FasterXML/jackson-dataformats-text/commit/e5dc40f55321161c94b4d1088a030cf9de936497

**Comment by sarutak** [comment]
Yeah, I also confirmed it depends on Jackson `2.13.0`.
https://mvnrepository.com/artifact/io.fabric8/kubernetes-client/5.9.0

**Comment by dongjoon-hyun** [comment]
Let me update it and run together to see the Jackson upgrade feasibility.

File: dev/deps/spark-deps-hadoop-2.7-hive-2.3 [hunk 1]
```diff
@@ -74,6 +74,7 @@ hadoop-annotations/2.7.4//hadoop-annotations-2.7.4.jar
 hadoop-auth/2.7.4//hadoop-auth-2.7.4.jar
 hadoop-aws/2.7.4//hadoop-aws-2.7.4.jar
 hadoop-azure/2.7.4//hadoop-azure-2.7.4.jar
+hadoop-client-api/3.3.1//hadoop-client-api-3.3.1.jar
```
Comment (dongjoon-hyun) [inline dev/deps/spark-deps-hadoop-2.7-hive-2.3:77]: Surprisingly, when I update Jackson, this appears. I will investigate this separately.

**Comment by dongjoon-hyun** [comment]
I spun off the Jackson part to SPARK-37241 .
- https://github.com/apache/spark/pull/34516

**Comment by dongjoon-hyun** [comment]
All test passed except 2 K8s integration test cases.
```
KubernetesSuite:
- Run SparkPi with no resources
- Run SparkPi with no resources & statefulset allocation
- Run SparkPi with a very long application name.
- Use SparkLauncher.NO_RESOURCE
- Run SparkPi with a master URL without a scheme.
- Run SparkPi with an argument.
- Run SparkPi with custom labels, annotations, and environment variables.
- All pods have the same service account by default
- Run extraJVMOptions check on driver
- Run SparkRemoteFileTest using a remote data file
- Verify logging configuration is picked from the provided SPARK_CONF_DIR/log4j.properties
- Run SparkPi with env and mount secrets.
- Run PySpark on simple pi.py example
- Run PySpark to test a pyfiles example
- Run PySpark with memory customization
- Run in client mode.
- Start pod creation from template
- PVs with local hostpath storage on statefulsets
- PVs with local hostpath and storageClass on statefulsets
- PVs with local storage
- Launcher client dependencies
- SPARK-33615: Launcher client archives
- SPARK-33748: Launcher python client respecting PYSPARK_PYTHON
- SPARK-33748: Launcher python client respecting spark.pyspark.python and spark.pyspark.driver.python
- Launcher python client dependencies using a zip file
- Test basic decommissioning
- Test basic decommissioning with shuffle cleanup
- Test decommissioning with dynamic allocation & shuffle cleanups *** FAILED ***
- Test decommissioning timeouts
- Run SparkR on simple dataframe.R example *** FAILED ***
Run completed in 33 minutes, 17 seconds.
Total number of tests run: 30
Suites: completed 2, aborted 0
Tests: succeeded 28, failed 2, canceled 0, ignored 0, pending 0
*** 2 TESTS FAILED ***
```

 This is not a regression because the master branch has it already (https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder-K8s/49466/)

**Comment by dongjoon-hyun** [comment]
The last commit is the same with the approved commit because Jackson upgrade is handled via another PR. I'll merge this, @sarutak .

**Pull request closed without merging**